Add support for building ZFS kernel modules

rn opened the pull request:
This adds support for building the ZFS kernel modules. The modules can be build as a separate package with tags matching the kernel build version. The modules can be added to a linuxkit image by adding the `zfs-kmod` package to the `init` section of a YAML file.

The `zfs-kmod` package also contains the standard kernel modules with `depmod` run over all of them so that `modprobe` should work.

The kernel modules do not compile for debug kernels due to CDDL incompatibility, apparently (the ZFS `./configure` script complains about `CONFIG_DEBUG_LOCK_ALLOC`)

Note, the `zfs-kmod` package is not build as the standard kernel build and is *not* hosted under the `linuxkit` Hub org due licensing uncertainty. Potential users will have to build the package themselves. `./docs/kernels.md` contains some notes.

/cc @matjohn2 (re #2333)

![piglet](https://user-images.githubusercontent.com/3338098/29937959-18d71a54-8e7f-11e7-8e8e-d6ca1fadd9fa.jpg)

justincormack commented:
We could make a "runtime `depmod`" setup - it could mount an overlay for example. Not sure if this is the best option. 

Couldnt you run `depmod` after building the zfs modules again? It would need to unpack the built modules and run `depmod` again, then copy the output of that to the final output which would then override the default. That doesnt work so well if you want to add several sets of kernel module addons though.

rn commented:
@justincormack I can certainly include ZFS in the standard kernel package and then depmod would be run. This PR create a seperate `zfs-kmod` package. I'm fine with including the ZFS kmods in the standard kernel package as well, if we decided to do so.

Runnning `depmod` takes some time so I don't really want to run it on boot with a suitable overlay by default

justincormack commented:
@rn no, I meant that this package could get all the modules from the build and then run depmod on the existing modules plus the zfs modules, then include the resulting files in its build, which will override the default ones.

I don't think including zfs in the default build is a good idea, in case people have legal concerns and want to totally ignore zfs.

rn commented:
Ah, I hadn't thought about this approach, but sounds doable. Will try next week when back.

rn commented:
OK, I've updated the commits:
- Don't build/push ZFS packages as part of the standard kernel build. Users have to build themselves.
- Add standard kernel modules to the package and run `depmod` so that `modprobe` should work
- Added some notes to the documentation

I've removed the `WIP/RFC` as I think this is ready for review/merge if folks are happy
